dm/syncer/checkpoint_flush_worker.go

```diff
@@ -0,0 +1,116 @@
+// Copyright 2021 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package syncer
+
+import (
+	"fmt"
+
+	"github.com/pingcap/tidb-tools/pkg/filter"
+	"go.uber.org/atomic"
+	"go.uber.org/zap"
+
+	tcontext "github.com/pingcap/ticdc/dm/pkg/context"
+	"github.com/pingcap/ticdc/dm/pkg/terror"
+)
+
+type checkpointFlushTask struct {
+	snapshotInfo *SnapshotInfo
+	// extra sharding ddl sqls
+	exceptTables  []*filter.Table
+	shardMetaSQLs []string
+	shardMetaArgs [][]interface{}
+	// async flush job
+	asyncflushJob *job
+	// error chan for sync flush
+	syncFlushErrCh chan error
+}
+
+type checkpointFlushWorker struct {
+	input        chan *checkpointFlushTask
+	cp           CheckPoint
+	execError    *atomic.Error
+	afterFlushFn func(task *checkpointFlushTask) error
+	addCountFunc func(bool, string, opType, int64, *filter.Table)
+}
+
+// Add add a new flush checkpoint job.
+func (w *checkpointFlushWorker) Add(msg *checkpointFlushTask) {
+	w.input <- msg
+}
+
+// Run read flush tasks from input and execute one by one.
+func (w *checkpointFlushWorker) Run(ctx *tcontext.Context) {
```

Comment:
ctx only use for log, how about use log.WithField like other components? https://github.com/pingcap/ticdc/blob/32032ca513ddb7c40831b590f9826dbea2d416a8/dm/syncer/dml_worker.go#L75

Reply:
actually, ctx is needed here for `w.cp.FlushPointsExcept(ctx, task.snapshotInfo.id, task.exceptTables, task.shardMetaSQLs, task.shardMetaArgs)`